src/test/java/com/redhat/pantheon/model/api/SlingResourceTest.java

```diff
@@ -0,0 +1,292 @@
+import org.apache.sling.testing.mock.sling.junit5.SlingContext;
+import org.apache.sling.testing.mock.sling.junit5.SlingContextExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Calendar;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith({SlingContextExtension.class, MockitoExtension.class})
+class SlingResourceTest {
+
+    private final SlingContext slingContext = new SlingContext();
+
+    @Test
+    public void fieldMapping() throws Exception {
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:name", "my-name",
+                        "jcr:date", Calendar.getInstance(),
+                        "jcr:number", "10")
+                .commit();
+
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        assertEquals("my-name", model.NAME.get());
+        assertEquals(new Long(10), model.NUMBER.get());
+        assertNotNull(model.DATE.get());
+    }
+
+    @Test
+    public void childMapping() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1")
+                .resource("child")
+                .resource("grandchild")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        model.CHILD
+                .get()
+                .GRANDCHILD
+                .get();
+    }
+
+    @Test
+    public void getDeepField() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1")
+                .resource("child")
+                .resource("grandchild",
+                        "jcr:name", "grandchild-name")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        assertEquals("grandchild-name", model.GRANDCHILD_NAME.get());
+    }
+
+    @Test
+    public void fieldEditing() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.MONTH, 1);
+        cal.set(Calendar.DATE, 1);
+        cal.set(Calendar.YEAR, 2019);
+        model.DATE.set(cal);
+        model.NAME.set("someoneelse");
+        model.NUMBER.set(15L);
```

Comment:
Did we need to make any other call to commit the changes to `model` before our assertions? If so, that might be a good idea to broaden the scope of the test a bit.

Reply:
No need to have a commit call as all `get` and `set`effectively operate on the underlying `Resource` object themselves. We are just testing that the fileds make it to that object as they should.